### PR TITLE
added a toast when the current location isnot working

### PR DESCRIPTION
**Pull request by Hambrsoom**
if the user clicks on my current location from the search page and the user's current location is not available then display a toast.

### CODE REVIEW

**Comment by viveanban** [comment]
Should the toast also be displayed when the location is not available when we click on the icon on the map?
image

File: app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java [hunk 1]
```diff
@@ -100,8 +100,8 @@ private void acceptOpenFromCalendarIntent() {
             RoomModel room = appDb.roomDao().getRoomByRoomCodeAndFloorCode(roomCode, floorCode);
 
             if(room!=null){
-                SelectingToFromState.setMyCurrentLocation(getMyCurrentLocation());
-                openRoutesPage(room, this);
+                SelectingToFromState.setMyCurrentLocation(currentLocation.getMyLocation());
+ //               openRoutesPage(room, this);
```
Comment (viveanban) [inline app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java:104]: why commented out?

File: app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java [hunk 2]
```diff
@@ -100,8 +100,8 @@ private void acceptOpenFromCalendarIntent() {
             RoomModel room = appDb.roomDao().getRoomByRoomCodeAndFloorCode(roomCode, floorCode);
 
             if(room!=null){
-                SelectingToFromState.setMyCurrentLocation(getMyCurrentLocation());
```
Comment (thomasgauvin) [inline app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java:103]: This change doesn't do anything

Comment (thomasgauvin) [reply app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java:103]: getMyCurrentLocation() was already doing that

File: app/src/main/java/com/example/concordia_campus_guide/helper/CurrentLocation.java [hunk 3]
```diff
@@ -45,9 +45,9 @@ public void run() {
                         setMyLocation(location);
                     }
                 });
-                handler.postDelayed(this, 2000);
+                handler.postDelayed(this, 500);
```
Comment (thomasgauvin) [inline app/src/main/java/com/example/concordia_campus_guide/helper/CurrentLocation.java:48]: Has anyone checked to see if this has performance issues?

File: app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java (hunk 1 above)
Comment (Hambrsoom) [reply app/src/main/java/com/example/concordia_campus_guide/activities/MainActivity.java:104]: I will put it back this here good catch

File: app/src/main/java/com/example/concordia_campus_guide/helper/CurrentLocation.java (hunk 3 above)
Comment (Hambrsoom) [reply app/src/main/java/com/example/concordia_campus_guide/helper/CurrentLocation.java:48]: When I tired it, there wasn't any issues. I asked Viveka and Ferdousara to try it too

Comment (viveanban) [reply app/src/main/java/com/example/concordia_campus_guide/helper/CurrentLocation.java:48]: I tested out the app on an emulator and I am not having any performance issues.

**Pull request merged**